tools/etcd-dump-logs/raw.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/zap"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/client/pkg/v3/fileutil"
+	"go.etcd.io/etcd/pkg/v3/pbutil"
+	"go.etcd.io/etcd/server/v3/storage/wal"
+	"go.etcd.io/etcd/server/v3/storage/wal/walpb"
+	"go.etcd.io/raft/v3/raftpb"
+)
+
+func readRaw(lg *zap.Logger, fromIndex *uint64, waldir string, out io.Writer) {
+	var walReaders []fileutil.FileReader
+	files, err := ioutil.ReadDir(waldir)
+	if err != nil {
+		lg.Fatal("Failed to read directory.", zap.String("directory", waldir), zap.Error(err))
+	}
+	for _, finfo := range files {
+		if filepath.Ext(finfo.Name()) != ".wal" {
+			lg.Warn("Ignoring not .wal file", zap.String("filename", finfo.Name()))
+		}
+		f, err := os.Open(filepath.Join(waldir, finfo.Name()))
+		if err != nil {
+			lg.Fatal("Failed to read file", zap.String("filename", finfo.Name()), zap.Error(err))
+		}
+		walReaders = append(walReaders, fileutil.NewFileReader(f))
+	}
+	decoder := wal.NewDecoderAdvanced(true, walReaders...)
+	// The variable is used to not pollute log with multiple continuous crc errors.
+	crcDesync := false
+	for {
+		rec := walpb.Record{}
+		err := decoder.Decode(&rec)
+		if err == nil || errors.Is(err, walpb.ErrCRCMismatch) {
+			if err != nil && !crcDesync {
+				lg.Warn("Reading entry failed with CRC error", zap.Error(err))
+				crcDesync = true
+			}
+			printRec(lg, &rec, fromIndex, out)
+			if rec.Type == wal.CrcType {
+				decoder.UpdateCRC(rec.Crc)
+				crcDesync = false
+			}
+			continue
+		}
+		if errors.Is(err, io.EOF) {
+			lg.Info("EOF: All entries were processed")
```

Comment:
I see that `log`, `zap.Logger `and `fmt` are used to output messages in this PR. Suggest to keep it consistent with the existing behavior:  Use `fmt` to output normal messages and `log` to output error or warning messages.

Reply:
Fixed.

Reply:
There are also some other places which need updated. Please search "lg." in `main.go `and `raw.go`.

Reply:
I thought log & lg  is long term the same logging mechanism as we do in etcd. 
Migrated to log for now.